### DISCUSSION TITLE
[Ethereum Provider] Some of the required chains are not supported

### DISCUSSION THREAD

**Discussion by sebastiancordea97** [WalletConnect/walletconnect-monorepo#3412] [Developer Support] [answered]
Hello!
This is my current config of EthereumProvider in Angular.

```
 private async initEthereumProvider() {
    this.ethereumProvider = await EthereumProvider.init({
      projectId: this.projectId, // REQUIRED your projectId
      chains: [1], // REQUIRED chain ids
      showQrModal: true, // REQUIRED set to "true" to use @walletconnect/modal
      methods: [
        "eth_sendTransaction",
        "eth_signTransaction",
        "eth_sign",
        "personal_sign",
        "eth_signTypedData",
        "eth_requestAccounts",
      ], // REQUIRED ethereum methods
      optionalMethods: [
        "eth_signTypedData",
        "eth_signTypedData_v4",
        "eth_sign",
      ], // OPTIONAL ethereum methods
      events: ["chainChanged", "accountsChanged", "connect", "disconnect"], // REQUIRED ethereum events
      qrModalOptions: {
        themeMode: "dark",
        explorerRecommendedWalletIds: [
          "1ae92b26df02f0abca6304df07debccd18262fdf5fe82daa81593582dac9a369",
          "4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0",
          "c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96",
          "c03dfee351b6fcc421b4494ea33b9d4b92a984f87aa76d1663bb28705e95034a",
          "c286eebc742a537cd1d6818363e9dc53b21759a1e8e5d9b263d0c03ec7703576",
          "8a0ee50d1f22f6651afcae7eb4253e52a3310b90af5daef78a8c4929a9bb99d4",
          "19177a98252e07ddfc9af2083ba8e07ef627cb6103467ffebb3f8f4205fd7927",
          "163d2cf19babf05eb8962e9748f9ebe613ed52ebf9c8107c9a0f104bfcf161b3",
          "4d0cf1b635a59175b4ad6d6522b0b748ee920b1f8c32030fa704c00926efdf3e",
          "fd20dc426fb37566d803205b19bbc1d4096b248ac04548e3cfb6b3a38bd033aa",
          "a797aa35c0fadbfc1a53e7f675162ed5226968b44a19ee3d24385c64d1d3c393",
          "971e689d0a5be527bac79629b4ee9b925e82208e5168b733496a09c0faed0709",
        ],
        explorerExcludedWalletIds: "ALL",
      },
    });

    this.ethereumProvider.on("connect", () => {});
    this.ethereumProvider.on("disconnect", () => {
      const selectedProvider = localStorage.getItem("selectedProvider");
      if (
        selectedProvider &&
        selectedProvider === SupportedWalletProviders.walletConnect
      ) {
        localStorage.removeItem("selectedProvider");
      }
      this.accountChangeSubject.next([]);
    });
  }

try {
      await this.initEthereumProvider();
      if (!this.hasOpenConnection) {
        await this.ethereumProvider.connect();
      }
      return await this.getWalletAddress();
    } catch (error) {
      console.error(error);
      return null;
    }

 public async getWalletAddress(): Promise<string> {
    const walletAddress = await this.ethereumProvider?.request({
      method: "eth_accounts",
    });
    return walletAddress[0];
  }


 "@walletconnect/ethereum-provider": "2.9.2",
 "@walletconnect/modal": "2.6.1",
 "ethers": "5.4.0",
```

Metamask connection works no problem
TrustWallet : browser error: User disapproved requested chains
                      mobile error: Some of the requested chains are not supported
Also Uniswap wallet has similar behaviour 

I saw this :https://lab-walletconnect-modal.pages.dev/with-ethereum-provider that works but i think is made on react or something.
I followed this guide: https://docs.walletconnect.com/2.0/web/providers/ethereum

What i am doing wrong in the config? Or shoud i also set:
walletCon.connect().then(() => {
      setProvider(new ethers.providers.Web3Provider(walletCon));
      setAddress(provider.accounts[0]);
    });

**Comment by glitch-txs** [comment, marked as answer]
Probably one of the methods you're using as required is not supported by the wallet and you'd want to move them to optionalMethod, same with events :)

**Reply by sebastiancordea97** [reply]
I'll try but is saying something about chains. I'll come back with an answer. Thanks

**Reply by sebastiancordea97** [reply]
Yes. I was overloading the config with crap i don't use. Thanks.

**Reply by jjzhang1** [reply]
How was it solved? Can you describe it in detail? I had the same problem

**Reply by sebastiancordea97** [reply]
@jjzhang1 Yes, of course. That is a generic error that has nothing related to chains. The problem was that some wallets do not support some methods in the "methods" array and also some events in the 'events' array. so the solutions was to move them to the optionalMethods and optionalEvents until i got a stable config for all wallets. Something like:
 projectId: this.projectId,
      chains: [1],
      optionalChains: [1, 5, 56, 42161, 11155111, 137],
      showQrModal: true,
      methods: ["personal_sign"],
      optionalMethods: [
        "personal_sign",
        "eth_sign",
        "eth_signTransaction",
        "eth_sendTransaction",
        "eth_signTypedData",
        "eth_signTypedData_v3",
        "eth_signTypedData_v4",
        "wallet_addEthereumChain",
        "eth_requestAccounts",
        "eth_accounts",
        "wallet_switchEthereumChain",
      ],
      optionalEvents: [
        "connect",
        "chainChanged",
        "accountsChanged",
        "disconnect",
      ],
 feel free to experiment with your own config. But the main idea is to not overload the "mandatory" array with stuff you do not use
